### ISSUE TITLE
Add display name for source metadata collection to entity

### ISSUE THREAD

**Issue by mlottesen** [odpi/egeria#448]
This gives consumers of an entity a way to display where an asset came from in a friendlier manner than just using the source metadata collection ID.

**Comment by mandy-chessell** [comment]
The easiest wat to consume the friendly name is to embed it into the instance's audit header.  The new field is called metadataCollectionName in class InstanceAuditHeader.

This field is populated with the local metadata collection name when the instance is created.
It is extracted from the event payload when an instance is being imported from the cohort.
It is extracted from the open metadata archive when the instances are loaded into a repository.

The name value used is under the control of the instance creator.  Egeria will use the following conventions:
  * Instances from an open metadata archive will use the archive name.
  * Instances from an external source will use a name determined by the OMAS that creates the instance (but default is qualifiedName of the SoftwareServerCapability instance for the external source.
  * Repositories will default to using the local server name.  This can be over-ridden with a new config variable.

When a repository starts up with a local repository enabled, message OMRS-AUDIT-0003 is logged.
In the example below, server cocoMDS7 does not have a metadata collection name set so it has defaulted to the server name.

cocoMDS7 OMRS-AUDIT-0003 The local repository is initializing with metadata collection named cocoMDS7 with an id of d8dd02bc-5cae-4c7d-8fc7-4f152ea4f503

Server cocoMDS1 has had the metadata collection name set to Freddy.

cocoMDS1 OMRS-AUDIT-0003 The local repository is initializing with metadata collection named Freddy with an id of 288a625b-1d7f-4313-b1c0-03a06904f7e4

The way to override the metadata collection name is through the following admin command.

```
POST <urlRoot>/open-metadata/admin-services/users/{userId}/servers/{serverName}/local-repository/metadata-collection-name/{name}
```

For example:
POST http://localhost:8080/open-metadata/admin-services/users/garygeeke/servers/cocoMDS1/local-repository/metadata-collection-name/Freddy

Other changes that were necessary include:
 * the reHomeEntity() and reHomeRelationship need the metadata collection name passed as a parameter.  I have added 2 new methods with this additional parameter and deprecated the old methods.  OMRSRepositoryConnector has both sets of methods implemented and so a repository connector that has implemented the old methods will work as well as one that has been migrated to the new.  Repository connectors checked into egeria have been migrated.
* The logic in the local repository connector wrapper that acts as a protective envelop to a real repository connector was changed to set the metadata collection name and id of instance comming out of the local repository if they have not been set.
* The registration events now pass the metadata collection name so other repositories can build a mapping of names to ids if they wish.  These names are stored in the cohort registry.
